metrics/web/lib/auth/domain/usecases/update_user_profile_usecase.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:meta/meta.dart';
+import 'package:metrics/auth/domain/repositories/user_repository.dart';
+import 'package:metrics/auth/domain/usecases/parameters/user_profile_param.dart';
+import 'package:metrics/base/domain/usecases/usecase.dart';
+
+/// A [UseCase] that provides an ability to update the user profile.
+@immutable
+class UpdateUserProfileUseCase
+    implements UseCase<Future<void>, UserProfileParam> {
+  /// A [UserRepository] provides an ability to interact
+  /// with the persistent store.
```

Comment:
```suggestion
  /// A [UserRepository] that provides an ability to interact
  /// with the persistent store.
```
Also, should we document repository in such a way in other use cases above? Seems like this variant is better.